include/grpc++/impl/codegen/async_stream.h

```diff
@@ -52,11 +52,14 @@ class ClientAsyncStreamingInterface {
 
   /// Request notification of the reading of the initial metadata. Completion
   /// will be notified by \a tag on the associated completion queue.
+  /// This call is optional, but if it is used, it cannot be used concurrently
+  /// with Read
   ///
   /// \param[in] tag Tag identifying this request.
   virtual void ReadInitialMetadata(void* tag) = 0;
 
-  /// Request notification completion.
+  /// Indicate that the stream is to be finished and request notification
+  /// Should not be used concurrently with other operations
```

Comment:
Since these are bidi streams, client-side finish isn't really like a read op; it really indicates the end. We don't view it as meaningful to write in parallel with finish. Finish is supposed to be called after read failure has happened.